controllers/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -71,9 +75,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		logger.Error(err, "unable to insert a new entry in status.updates after NSTemplateTier changed")
 		return reconcile.Result{}, errs.Wrap(err, "unable to insert a new entry in status.updates after NSTemplateTier changed")
 	} else if added {
-		logger.Info("Requeing after adding a new entry in tier.status.updates")
+		logger.Info("Requeue after adding a new entry in tier.status.updates")
 		return reconcile.Result{Requeue: true}, nil
 	}
+
+	// check if the `status.revisions` field is up-to-date and create a TTR for each TierTemplate
+	if created, err := r.ensureRevision(ctx, tier); err != nil {
+		return reconcile.Result{}, errs.Wrap(err, "unable to create new TierTemplateRevision after NSTemplateTier changed")
+	} else if created {
+		logger.Info("Requeue after creating a new TTR")
```

Comment:
I thought about that but how do we map TTRs to nstemplatetier ? Since as you correctly pointed out in [this comment](https://github.com/codeready-toolchain/host-operator/pull/1103#discussion_r1842286283) , TierTemplate ( and thus TTRS ) can be used by multiple NSTemplateTiers.

Reply:
Ok, good point! I can do that. Am also already setting those lables on TTRs , but I need to use the tierName from the NSTemplateTier and not from the TierTemplate when setting them.

Thanks for the suggestion 🙏

Reply:
I've added the watcher for TTRs in https://github.com/codeready-toolchain/host-operator/pull/1103/commits/9d3e46acb58393fa3e0b3381916a9f6d9cf940a5

Reply:
I had to revert the watcher changes in https://github.com/codeready-toolchain/host-operator/pull/1103/commits/e8ecb9dcd2f7107fddacdf9a015cbb905aeb8f31 that was generating copies of the TTRs since the creation of the TTR was faster than the update of the NSTemplateTier.Status.Revisions field.